### PR TITLE
#187419032 A buyer should be able to track their order status

### PR DESCRIPTION
**Pull request by Hakizimana-Clement**
### What does this PR do?
A buyer should be able to track their order status.

### How should this be manually tested?
1. Clone the repo and cd into it
2. Checkout to ```ft-buyer-tracking-order-187419032``` and ```run npm install```
3. Run npm run dev to start the application locally
4.  Click   ```My Orders``` in Navbar and  see all order information, even single order.

### Pivot tracker Id
#187419032

### CODE REVIEW

File: src/assets/EmptyOrder.png
Comment (GarrixA) [inline src/assets/EmptyOrder.png:1]: Add images in Image utils

File: src/assets/order-icon.svg
Comment (GarrixA) [inline src/assets/order-icon.svg:1]: Add this in Image utils too

File: src/components/dashboard/buyer/OrderTableRow.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	return (
+		<tr
+			key={idx}
+			className={`relative text-sm ${idx % 2 !== 0 ? 'bg-[#DDD]' : ''}`}
+		>
+			<td className="text-sm font-bold">
+				#{item?.id.length > 20 ? item.id?.slice(0, 5) : item.id}
+			</td>
+			<td>{new Date(firstOrderData?.deliveryDate).toLocaleDateString()}</td>
```
Comment (Tuyisenge2) [inline src/components/dashboard/buyer/OrderTableRow.tsx]: date in table order Date column is not same as the data orders was requested check on it and make it accurate 

and when user navigate to view more of order , the date of delivery is becoming the date of Order Date

**Review by GarrixA — CHANGES_REQUESTED** [review]
Screenshot 2024-07-18 at 19 34 51

In small devices I cannot go back. the button to go back is not working

File: src/components/cards/orderModel.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+export const modalContext = createContext({});
+export const OrderModel = ({ id, idx, handleToggle }: OrderModalProps) => {
+	return (
+		<div className="absolute desktop:w-[15%] right-2 laptop:right-6 laptop:pr-4 laptop:pt-6 z-50 -top-10 flex p-2 rounded-lg shadow-md bg-[#fcfdfe] _shadow pt-5 pl-5">
+			<div className="flex flex-col justify-between relative w-full">
+				<div className="view_product flex gap-2 text-sm rounded-md p-1 a_link mt-5 mb-3">
+					<Link to={`/orders/${id}`} className="flex gap-2">
+						<ShoppingBasket className="text-sm" /> View More
```
Comment (Tuyisenge2) [inline src/components/cards/orderModel.tsx]: improve the design of view more here by centering view more text

**Review by Tuyisenge2 — CHANGES_REQUESTED** [review]

File: src/assets/EmptyOrder.png
Comment (Hakizimana-Clement) [reply src/assets/EmptyOrder.png:1]: Its always there.

File: src/assets/order-icon.svg
Comment (Hakizimana-Clement) [reply src/assets/order-icon.svg:1]: Its already there.

File: src/components/cards/orderModel.tsx (hunk 2 above)
Comment (Hakizimana-Clement) [reply src/components/cards/orderModel.tsx]: thx, let me fix it.

File: src/components/dashboard/buyer/OrderTableRow.tsx (hunk 1 above)
Comment (Hakizimana-Clement) [reply src/components/dashboard/buyer/OrderTableRow.tsx]: Thx, let me fix it

File: src/components/cards/orderModel.tsx (hunk 2 above)
Comment (Tuyisenge2) [reply src/components/cards/orderModel.tsx]: still i am seeing this issues
Screenshot 2024-07-20 095058

Comment (Hakizimana-Clement) [reply src/components/cards/orderModel.tsx]: I solve it.

**Review by GarrixA — APPROVED** [review]

**Review by Tuyisenge2 — APPROVED** [review]

File: src/components/Nav.tsx [hunk 3]
```diff
@@ -92,19 +93,21 @@ const Nav = () => {
...
-											data-testid="number-of-item"
-											className="bg-neutral-darkRed text-sm mobile:text-base p-1 absolute -top-3 -right-3 rounded-full flex items-center justify-center text-neutral-white font-semibold mobile:w-6 w-5 h-5 mobile:h-6"
-										>
-											{numberOfItem}
-										</span>
-									) : null}
-								</div>
-							</Link>
+							{fetchInfo()?.role === 'BUYER' && (
+								<Link to={'/carts'}>
```
Comment (leandreAlly) [inline src/components/Nav.tsx:97]: let's not fetch info in UI part, we can fetch them while component is rendering.

like const userData = fetchInfo()

userData.role ...... the we perform the task

File: src/components/ProfileDropdown.tsx [hunk 4]
```diff
@@ -82,14 +82,16 @@ const ProfileDropdown = ({ image }: ProfileDropdownProps) => {
...
 						</Link>
-						<Link
-							to="/orders"
-							className="block px-4 py-2 text-sm text-gray-700 hover:bg-primary-lightblue hover:text-neutral-white"
-							role="menuitem"
-							onClick={toggleDropdown}
-						>
-							My orders
-						</Link>
+						{fetchInfo()?.role === 'BUYER' && (
```
Comment (leandreAlly) [inline src/components/ProfileDropdown.tsx]: same case here let's not fetch info in UI part

**Review by leandreAlly — CHANGES_REQUESTED** [review]

File: src/components/Nav.tsx (hunk 3 above)
Comment (Hakizimana-Clement) [reply src/components/Nav.tsx:97]: ooh i understand, let me fix it

File: src/components/ProfileDropdown.tsx (hunk 4 above)
Comment (Hakizimana-Clement) [reply src/components/ProfileDropdown.tsx]: Thanks, let me fix i

**Pull request merged**